documentation/modules/oauth/con-oauth-authentication-broker.adoc

```diff
@@ -0,0 +1,138 @@
+
+== {oauth} authentication configuration in the Kafka cluster
+
+To use {oauth} authentication in the Kafka cluster, you specify a TLS listener configuration for your Kafka cluster custom resource with the authentication method `oauth`:
+
+.Authentication method type for {oauth}
+[source,yaml,subs="+quotes, attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+spec:
+  kafka:
+    listeners:
+      *tls*:
+        authentication:
+          type: *oauth*
+          #...
+----
+
+You can configure `plain`, `tls` and `external` listeners, as described in xref:assembly-configuring-kafka-broker-listeners-deployment-configuration-kafka[Kafka broker listeners],
+but it is recommended not to use `plain` listeners with {oauth} as this creates a vulnerability to network eavesdropping and unauthorized access through token theft.
+
+You configure an `external` listener with `tls: true` for a secure transport layer to communicate with the client.
+
+.External listener with TLS
+[source,yaml,subs="+quotes"]
+----
+# ...
+listeners:
+  tls:
+    authentication:
+      type: oauth
+  external:
+    type: oauth
+    *tls: true*
+    authentication:
+      type: tls
+# ...      #...
+----
+
+== Fast local JWT token validation configuration
+
+Fast local JWT token validation checks a JWT token signature locally.
+
+The local check ensures that a token:
+
+* Conforms to type by containing a (_typ_) claim value of `Bearer` for an access token
+* Is valid (not expired)
+* Has an issuer that matches a `validIssuerURI`
+
+You specify a `validIssuerUrI` attribute when you configure the listener, so that any tokens not issued by the authorization server are rejected.
+
+The authorization server does not need to be contacted during fast local JWT token validation.
+You activate fast local JWT token validation by specifying a `jwksEndpointUri` attribute, the endpoint exposed by the {oauth} authorization server.
+The endpoint contains the public keys used to validate signed JWT tokens, which are sent as credentials by Kafka clients.
+
+NOTE: All communication with the authorization server should be performed using TLS encryption.
+
+You can configure a certificate truststore as a Kubernetes Secret in your {ProductName} project namespace, and use a `tlsTrustedCertificates` attribute to point to the Kubernetes Secret containing the truststore file.
+
+You might want to configure a `userNameClaim` to properly extract a username from the JWT token.
+If you want to use Kafka ACL authorization, you will need to identify the user by their username during authentication.
+(The `sub` claim in JWT tokens is typically a unique ID, not a username.)
+
+.Example configuration for fast local JWT token validation
+[source,yaml,subs="+quotes, attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+spec:
+  kafka:
+    listeners:
+      tls:
+        authentication:
+          type: oauth
+          validIssuerUri: _<URL-on-auth-server>_
```

Comment:
Can we get here some real examples instead of the place holders? In the other places you use for example `validIssuerUri: <__https://<auth-server-address>/auth/realms/external__>` which is great.

Reply:
These URLs depend on the specific authorization server, and how it is configured. Maybe if there is a section in the docs with specific examples using a specific authorization server it would have to contain fully working config examples that can be copy pasted. Maybe here we don't really want that?

Reply:
Seeing the example address might help the users better understand what kind of URL are we looking for (at least in some cases). `URL-on-auth-server` doesn't tell them anything.

Reply:
I've based the examples on the external one we use in the procedure. But as we're not using external here, I swapped 'external' in the path to `tls`.